cloud/amazon/redshift.py

```diff
@@ -0,0 +1,938 @@
+        if is_not_found_json_response_error(e):
+            return False
+
+    return bool(resource)
+
+
+def check_resource_status(redshift, identifier, resource_type='cluster',
+                          desired_status='available'):
+    """
+    Return True if the cluster has the given desired status, False otherwise
+    """
+    has_status = False
+
+    resource = None
+    status_field = None
+
+    if resource_type == 'cluster':
+        fetch = get_cluster
+        status_field = 'ClusterStatus'
+    elif resource_type == 'snapshot':
+        fetch = get_snapshot
+        status_field = 'Status'
+    else:
+        raise ValueError(
+            'Unrecognized resource type: {}'.format(resource_type))
+    try:
+        resource = fetch(redshift, identifier)
+    except boto.exception.JSONResponseError, e:
+        if is_not_found_json_response_error(e):
+            pass
+
+    if not resource:
+        pass
+
+    if resource and (resource.get(status_field) == desired_status):
+        has_status = True
+
+    return has_status
+
+
+def check_snapshot_exists(redshift, snapshot_identifier):
+    """
+    Return True if the given snapshot exists, False otherwise
+    """
+    return check_resource_exists(redshift, snapshot_identifier,
+                                 resource_type='snapshot')
+
+
+def wait_for_condition(condition_check, timeout):
+    """
+    Given a callable representing a check for some condition, wait until the
+    condition is met, within some maxiumum amount of time (timeout).
+
+    The given condition_check callable is expected to return True when the
+    condition is met, False otherwise.
+
+    Timeout is the maxiumum number of seconds to wait.
+
+    If timeout is set to 0, then we will wait forever.
+
+    Returns:
+    - True: condition was met within the timeout period
+    - False: reached timeout and the condition was still not met
+    """
+    wait_timeout = time.time() + timeout
+
+    if timeout == 0:
+        def time_check(): return True
+    else:
+        def time_check(): return wait_timeout > time.time()
+
+    while time_check():
+        if condition_check():
+            return True
+        time.sleep(5)
+
+    return False
+
+
+def json_response_err_msg(json_response_error):
+    """ Given a JSONResponseError from boto, return the error message. """
+    err_msg = None
+
+    if hasattr(json_response_error, 'body') and json_response_error.body:
+        if type(json_response_error.body) in types.StringTypes:
+            err_msg = json_response_error.body
+        elif isinstance(json_response_error.body, dict):
+            err_msg = json_response_error.body.get('Error', {}).get('Message')
+
+    if not err_msg:
+        err_msg = str(json_response_error)
+    return err_msg
+
+
+def main():
+    global CHECK_MODE
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update({
+
+        'command':                      dict(choices=['create',
```

Comment:
Please don't do this. It's horrible to read. Just follow the same formatting as every other module in Ansible